### PR TITLE
Add new admin APIs to remove media by media ID from quarantine.

### PR DESCRIPTION
**Pull request by dklimpel**
Related to: #6681, #5956, #10040

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

**Review by richvdh — COMMENTED** [review]
lgtm, though as with #10040 I question whether we should use `DELETE` or `POST .../unquarantine/...`.

Let's keep the discussion at #10040 for now.

File: synapse/storage/databases/main/room.py [hunk 1]
```diff
@@ -875,6 +899,46 @@ def _quarantine_media_txn(
...
+            """
+                UPDATE remote_media_cache
+                SET quarantined_by = ?
+                WHERE media_origin = ? AND media_id = ?
+            """,
+            ((None, origin, media_id) for origin, media_id in remote_mxcs),
+        )
+        total_media_remove_quarantine += txn.rowcount if txn.rowcount > 0 else 0
+
+        return total_media_remove_quarantine
```
Comment (dklimpel) [inline synapse/storage/databases/main/room.py:940]: I think the part for discussion in this PR is this. This is nearly a clone of `_quarantine_media_txn` only without parameter `quarantined_by` an a SQL query without `AND safe_from_quarantine = ?`
https://github.com/matrix-org/synapse/blob/c89da79155013b802899c2e5d3a53ba993972c22/synapse/storage/databases/main/room.py#L860-L900

I can change it to one function if you like. Then I will make parameter `quarantined_by` to `Optional[str]`

Comment (richvdh) [reply synapse/storage/databases/main/room.py:940]: well, I did wonder about that, but didn't want to insist that you change it.

Yes, I think that combining the functions would be an improvement, if you have time to do so. Thanks.

Comment (dklimpel) [reply synapse/storage/databases/main/room.py]: I have made an update.

**Review by richvdh — APPROVED** [review]
Thank you!

**Pull request merged**